common/functions/src/scalars/semi_structureds/array_get.rs

```diff
@@ -0,0 +1,143 @@
+            .features(FunctionFeatures::default().deterministic().num_arguments(2))
+    }
+}
+
+impl Function for ArrayGetFunction {
+    fn name(&self) -> &str {
+        &*self.display_name
+    }
+
+    fn return_type(&self) -> DataTypeImpl {
+        NullableType::new_impl(self.array_type.inner_type().clone())
+    }
+
+    fn eval(
+        &self,
+        _func_ctx: FunctionContext,
+        columns: &ColumnsWithField,
+        input_rows: usize,
+    ) -> Result<ColumnRef> {
+        let array_column: &ArrayColumn = if columns[0].column().is_const() {
+            let const_column: &ConstColumn = Series::check_get(columns[0].column())?;
+            Series::check_get(const_column.inner())?
+        } else {
+            Series::check_get(columns[0].column())?
+        };
+
+        let inner_type = self.array_type.inner_type().data_type_id();
+        let index_type = columns[1].data_type().data_type_id();
+        with_match_scalar_types_error!(inner_type.to_physical_type(), |$T1| {
+            with_match_physical_primitive_type_error!(index_type.to_physical_type(), |$T2| {
+                let inner_column: &<$T1 as Scalar>::ColumnType = Series::check_get(array_column.values())?;
+                let mut builder = NullableColumnBuilder::<$T1>::with_capacity(input_rows);
+                if columns[0].column().is_const() {
+                    let index_column: &PrimitiveColumn<$T2> = if columns[1].column().is_const() {
+                        let const_column: &ConstColumn = Series::check_get(columns[1].column())?;
+                        Series::check_get(const_column.inner())?
+                    } else {
+                        Series::check_get(columns[1].column())?
+                    };
+                    let len = array_column.size_at_index(0);
+                    for i in 0..input_rows {
+                        let index = index_column.get(i).as_u64()? as usize;
+                        let _ = check_index(index, len)?;
+                        builder.append(inner_column.get_data(index), true);
+                    }
+                } else if columns[1].column().is_const() {
+                    let index_column: &ConstColumn = Series::check_get(columns[1].column())?;
+                    let index = index_column.get(0).as_u64()? as usize;
+                    let mut offset = 0;
+                    for i in 0..input_rows {
+                        let len = array_column.size_at_index(i);
+                        let _ = check_index(index, len)?;
+                        builder.append(inner_column.get_data(offset + index), true);
+                        offset += len;
+                    }
+                } else {
+                    let index_column: &PrimitiveColumn<$T2> = Series::check_get(columns[1].column())?;
+                    let mut offset = 0;
+                    for i in 0..input_rows {
+                        let index = index_column.get(i).as_u64()? as usize;
```

Comment:
we should not use `get(i)` to construct a datavalue and convert it into `as_u64`


```
for (row, index) in index_column.iter().enumerate() {
      if index < 0 { ...}
      
      ....

}
```

Reply:
ok